web_access_rule_buttons/models.py

```diff
@@ -28,4 +28,11 @@ def check_access_rule_all(self, operations=None):
     return result
 
 
-models.BaseModel.check_access_rule_all = check_access_rule_all
+# Could be any model, we just use a core model to have a 'register_hook'
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    def _register_hook(self, cr):
+        # Add method check_access_rule_all for all models
+        models.BaseModel.check_access_rule_all = check_access_rule_all
```

Comment:
Yes it could be another solution which would not involves to add a method on `BaseModel` I didn't thought of, on the other hand, the API would be a bit confusing as it would take model + ids of the other model (as the method is a `@api.multi`). It would also no longer be coherent with `check_access_rule`.

This is not a change I would do on a stable module, but as this one is recent, chances that someone relies on this method to be on `BaseModel` are low... could be okay.